ExtraDry/ExtraDry.Core/Models/TaxonomyEntity.cs

```diff
@@ -20,24 +20,32 @@ public void SetParent(T? parent)
             Ancestors.AddRange(parent.Ancestors);
             Ancestors.Add(parent);
         }
+        var self = this as T;
+        if(self != null) {
```

Comment:
`if(this is T self)` could more concisely replace these two lines...

Reply:
As i was writing it, I was thinking "I know there's a more concise way to do this" but I couldn't think of it, and becuase it wasn't that verbose, I didn't think i revisit. Good catch